### PR TITLE
[DE-234] Merge Composer and Elementary logs v2

### PR DESCRIPTION
**Pull request by jackwelty**
Alrighty let's try this again! We had to revert (#140) the previous version of this PR (#139) because the Composer logs table was _not_ quite as unique as expected (and dbt tests caught this!).

A TL;DR on what happened--when a DAG is re-run, this kicks off another `DagRun Finished` log, so we needed to add an intermediate step to smartly dedupe across these inserts. For most data points, we're just taking the field from the most recent run, but I tried to do some reasonable things around min/max dates, aggregating run time, etc.

Changes since the last time this PR was reviewed:
* `int_group_scheduler_logs_by_dag` added
* Tests on `int_parse_scheduler_logs` removed to reflect non-uniqueness of `dag_id`/`run_id`
* `int_merge_logs` now pulls from `int_group_scheduler_logs_by_dag`

A [green run](https://256d9e043bf64c2d9fda5c645b99c062-dot-us-central1.composer.googleusercontent.com/dags/dbt_monitoring/grid?dag_run_id=manual__2023-05-15T15%3A50%3A03.341285%2B00%3A00&task_id=dbt_monitoring_test_models) in dev Composer.

### CODE REVIEW

File: dbt-cta/monitoring/models/1_intermediate/composer_logs/_int_composer_logs__models.yml [hunk 1]
```diff
@@ -36,6 +36,45 @@ models:
...
+      - name: run_start_date
+      - name: run_end_date
+      - name: run_duration
+      - name: state
+        tests:
+          - accepted_values:
+              values: ['success', 'failed']
+      - name: external_trigger
+      - name: run_type
+      - name: data_interval_start
```
Comment (emily-flambe) [inline dbt-cta/monitoring/models/1_intermediate/composer_logs/_int_composer_logs__models.yml:75]: QQ - what are these `data_interval` fields about? Can we add a description to one of them to explain?

**Review by emily-flambe — DISMISSED** [review]
Had a question about a field but the code LGTM! Fire at will

**Review by kanelouise — COMMENTED** [review]
Just wanted to say I love how descriptive your PRs are, @jackwelty!

**Comment by jackwelty** [comment]
> Just wanted to say I love how descriptive your PRs are, @jackwelty!

thank you @kanelouise, I do my best! 💙

**Pull request merged**